public/app/plugins/panel/heatmap/utils.ts

```diff
@@ -772,6 +777,8 @@ export function heatmapPathsPoints(opts: PointsBuilderOpts, exemplarColor: strin
         let fillPaths = [points];
         let fillPalette = [exemplarColor ?? 'rgba(255,0,255,0.7)'];
 
+        let yShift = yLayout === HeatmapCellLayout.le ? -0.5 : yLayout === HeatmapCellLayout.ge ? 0.5 : 0;
```

Comment:
this fixes vertical exemplar positioning in each cell. previously we assumed `le` (cell down from tick), but with ordinal it is centered, and can be manually set as well via panel opts.